Library/Homebrew/cmd/deps.rb

```diff
@@ -82,85 +86,93 @@ class Deps < AbstractCommand
 
       sig { override.void }
       def run
+        raise UsageError, "`brew deps --os=all` is not supported" if args.os == "all"
+        raise UsageError, "`brew deps --arch=all` is not supported" if args.arch == "all"
+
+        os, arch = T.must(args.os_arch_combinations.first)
         all = args.eval_all?
 
         Formulary.enable_factory_cache!
```

Comment:
I double-checked to make sure that the caching strategy supports multiple platforms.